src/basic/env-util.c

```diff
@@ -779,7 +779,7 @@ int serialize_environment(FILE *f, char **environment) {
                 if (!ce)
                         return -ENOMEM;
 
-                fprintf(f, "env=%s\n", *e);
+                fprintf(f, "env=%s\n", ce);
```

Comment:
Well, it's because the escaping rules defined by the shell and by systemd unit files are different. It's too late to change. If anything, we can always add a slightly different way to parse environment variables in systemd config files (either a new option, or some magic syntax...).